test/source/allocation.cpp

```diff
@@ -19,13 +19,79 @@ auto get_one(smeagle::Corpus const& corpus, char const* name) {
   return funcs;
 }
 
-TEST_CASE("Register Allocation") {
+TEST_CASE("Register Allocation and Types") {
   auto corpus = smeagle::Smeagle("liballocation.so").parse();
 
   SUBCASE("one int") {
-    auto funcs = get_one(corpus, "f1");
+    auto funcs = get_one(corpus, "test_type_int");
     REQUIRE(funcs.size() == 1);
     auto const& parameters = funcs[0].parameters;
+    CHECK(parameters[0].type == "int");
+    CHECK(parameters[0].location == "%rdi");
+  }
+
+  SUBCASE("one string") {
+    auto funcs = get_one(corpus, "test_type_string");
+    REQUIRE(funcs.size() == 1);
+    auto const& parameters = funcs[0].parameters;
+    CHECK(parameters[0].type == "string");
+    CHECK(parameters[0].location == "%rdi");
+  }
+
+  SUBCASE("one char") {
+    auto funcs = get_one(corpus, "test_type_char");
+    REQUIRE(funcs.size() == 1);
+    auto const& parameters = funcs[0].parameters;
+    CHECK(parameters[0].type == "char");
+    CHECK(parameters[0].location == "%rdi");
+  }
+
+  SUBCASE("one bool") {
+    auto funcs = get_one(corpus, "test_type_boolean");
+    REQUIRE(funcs.size() == 1);
+    auto const& parameters = funcs[0].parameters;
+    CHECK(parameters[0].type == "bool");
+    CHECK(parameters[0].location == "%rdi");
+  }
+
+  SUBCASE("void") {
+    auto funcs = get_one(corpus, "test_type_void");
+    REQUIRE(funcs.size() == 1);
+    auto const& parameters = funcs[0].parameters;
+    REQUIRE(parameters.size() == 0);
+  }
+
+  SUBCASE("one wchar_t") {
+    auto funcs = get_one(corpus, "test_type_wchar_t");
+    REQUIRE(funcs.size() == 1);
+    auto const& parameters = funcs[0].parameters;
+    CHECK(parameters[0].type == "wchar_t");
+    CHECK(parameters[0].location == "%rdi");
+  }
+
+  SUBCASE("one float") {
```

Comment:
We don't correctly handle floating point types yet. I wouldn't have a test case for this- particularly since it should be in %xmm0, not %xmm7.

Reply:
Ah my loop is backwards.